openstack_hypervisor/hooks.py

```diff
@@ -1791,6 +1738,159 @@ def _configure_libvirt_tls(snap: Snap) -> None:
         ],
         PRIVATE_PERMS,
     )
+    _configure_webdav_tls(snap, cacert, cert, key)
+
+
+def _configure_webdav_tls(snap: Snap, cacert: bytes, cert: bytes, key: bytes) -> None:
+    """Configure TLS material for the Nova WebDAV fileserver."""
+    pki_root = snap.paths.common / Path("etc/pki")
+
+    # Nova fileserver TLS (server cert/key for WebDAV)
+    nova_dir = pki_root / Path("nova")
+    nova_private_dir = nova_dir / Path("private")
+
+    nova_cert = nova_dir / "servercert.pem"
+    nova_key = nova_private_dir / "serverkey.pem"
+    _template_tls_file(cert, nova_cert, [])
+    _template_tls_file(key, nova_key, [], PRIVATE_PERMS)
+
+    # Nova fileserver CA (for client verification)
+    nova_ca = nova_dir / "ca-cert.pem"
+    _template_tls_file(cacert, nova_ca, [])
+
+    common_base = snap.paths.common
+    tls_parent = common_base / Path("apache-webdav")
+    webdav_tls_dir = tls_parent / "tls"
+
+    apache_cert = webdav_tls_dir / "servercert.pem"
+    apache_key = webdav_tls_dir / "serverkey.pem"
+    apache_ca = webdav_tls_dir / "ca-cert.pem"
+
+    shutil.copyfile(nova_cert, apache_cert)
```

Comment:
Copying of these files must be handled with precaution, such as:
```python
def secure_copy(src: Path, dst: Path, mode: int = 0o600):
    tmp = dst.with_suffix('.tmp')
    tmp.touch(mode=mode)
    shutil.copyfile(src, tmp)
    shutil.chown(tmp, user="snap_daemon", group="snap_daemon")
    tmp.rename(dst)
```

(this is just an example at the top of my head, we must ensure security)